src/campaigns/mapping/mapping.service.ts

```diff
@@ -0,0 +1,221 @@
+
+const APP_BASE = process.env.CORS_ORIGIN as string
+const isProd = APP_BASE === 'https://goodparty.org'
+
+@Injectable()
+export class MappingService {
+  constructor(private readonly prisma: PrismaService) {}
+
+  async listMapCount(
+    stateFilter?: string,
+    resultsFilter?: boolean,
+  ): Promise<{ count: number }> {
+    const baseAndConditions = buildMapFilters({
+      stateFilter,
+      resultsFilter,
+      isProd,
+    })
+
+    const additionalAndConditions: Prisma.CampaignWhereInput[] = [
+      {
+        details: {
+          path: ['geoLocation', 'lng'],
+          not: { equals: null },
+        },
+      },
+      {
+        details: {
+          path: ['geoLocationFailed'],
+          equals: false,
+        },
+      },
+      {
+        OR: [
+          { didWin: true },
+          {
+            didWin: null,
+            details: {
+              path: ['electionDate'],
+              gte: sevenDaysAgo,
+            },
+          },
+        ],
+      },
+    ]
+
+    const combinedAndConditions: Prisma.CampaignWhereInput[] = [
+      ...baseAndConditions,
+      ...additionalAndConditions,
+    ]
+
+    const where: Prisma.CampaignWhereInput = {
+      userId: { not: null },
+      isDemo: false,
+      isActive: true,
+      AND: combinedAndConditions,
+    }
+    const count = await this.prisma.campaign.count({
+      where,
+    })
+
+    return { count }
+  }
+
+  async listMap(
+    partyFilter?: string,
+    stateFilter?: string,
+    levelFilter?: string,
+    resultsFilter?: boolean,
+    officeFilter?: string,
+    nameFilter?: string,
+    forceReCalc?: boolean,
+  ): Promise<CleanCampaign[]> {
+    const baseAndConditions = buildMapFilters({
+      partyFilter,
+      stateFilter,
+      levelFilter,
+      resultsFilter,
+      officeFilter,
+      isProd,
+    })
+
+    const additionalFilters: Prisma.CampaignWhereInput = {
+      OR: [
+        { didWin: true },
+        {
+          didWin: null,
+          details: {
+            path: ['electionDate'],
+            gte: sevenDaysAgo,
+          },
+        },
+      ],
+    }
+
+    const combinedAndConditions: Prisma.CampaignWhereInput[] = [
+      ...baseAndConditions,
+      additionalFilters,
+    ]
+
+    const where: Prisma.CampaignWhereInput = {
+      userId: { not: null },
+      isDemo: false,
+      isActive: true,
+      AND: combinedAndConditions,
+    }
+
+    if (nameFilter) {
+      where.user = {
+        AND: [
+          {
+            OR: [
+              { firstName: { contains: nameFilter, mode: 'insensitive' } },
+              { lastName: { contains: nameFilter, mode: 'insensitive' } },
+            ],
+          },
+        ],
+      }
+    }
+
+    const campaigns = await this.prisma.campaign.findMany({
+      where,
+      include: {
+        user: {
+          select: {
+            firstName: true,
+            lastName: true,
+            avatar: true,
+          },
+        },
+      },
+    })
```

Comment:
This should be going through the `CampaignsService`, not directly to the ORM.

```suggestion
    const campaigns = await this.campaignsService.findAll({
      where,
      include: {
        user: {
          select: {
            firstName: true,
            lastName: true,
            avatar: true,
          },
        },
      },
    })
```

Reply:
I guess I should change that for all my campaign prisma use then, because there's more than just that spot. I am a little confused though why our campaign service methods take different arguments than the original prisma method, doesn't work the best for what I'm trying to do in some spots, because I'm using Prisma's actual types

Reply:
Yeah.. def go ahead and refactor those to allow for us to have the same arguments interface to the Prisma model methods, if it makes sense to do so.